test/test_autograd.py

```diff
@@ -202,6 +202,15 @@ def compute_grad(create_graph):
         x_grad, x_grad_clone = compute_grad(create_graph=True)
         self.assertEqual(x_grad, x_grad_clone)
 
+    def test_sum_to_with_empty_dim_grad(self):
+        a = torch.rand(4, 0)
+        b = torch.rand(4, 1, requires_grad=True)
+        c = a + b
+        assert c.shape == (4, 0)
+        c.sum().backward()
+
+        self.assertEqual(b.grad, torch.zeros(4, 1))
```

Comment:
any reason you don't make `a` require grad and check its grad shape too?

Reply:
No reason. That's a good thing to check, I'll add it in